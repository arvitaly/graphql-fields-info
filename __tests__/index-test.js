"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const g = require("graphql");
const __1 = require("./..");
const schema_1 = require("./../__fixtures__/schema");
describe("info", () => {
    it("simple query and print", () => {
        const schema = new g.GraphQLSchema({
            query: new g.GraphQLObjectType({
                name: "Query",
                fields: {
                    test: { type: g.GraphQLString },
                },
            }),
        });
        const parser = __1.fromQuery(`query Q1{
            test
        }`, schema);
        expect(parser.getFields()).toMatchSnapshot();
        expect(parser.print()).toMatchSnapshot();
    });
    it("parse all fields", () => {
        const parser = __1.fromQuery(`query Q1{
        node{
            model1{ field1 }
            model2{
                ... on Model2{
                    field2
                }
            }
        }
    }`);
        expect(parser.getFields()).toMatchSnapshot();
    });
    it("get node fields", () => {
        const parser = __1.fromQuery(`query Q1{ node(id:"1"){ model1{ field1 } } }`);
        expect(parser.getNodeFields()).toMatchSnapshot();
    });
    it("get query one fields", () => {
        const parser = __1.fromQuery(`query Q1{ viewer{ model1{ field1, model2{ field2 } }} }`);
        expect(parser.getQueryOneFields()).toMatchSnapshot();
    });
    it("get query connection fields simple", () => {
        const parser = __1.fromQuery(`query Q1{
            viewer{
                modelName1s{
                    edges{
                        node{
                            name
                        }
                    }

                }
            }
        }`);
        expect(parser.getQueryConnectionFields()).toMatchSnapshot();
    });
    it("get query connection fields with fragments, change selection node and print", () => {
        const parser = __1.fromQuery(`query Q1{
            viewer{
                model1(first:10){
                    pageInfo{
                        hasNextPage
                    }
                    ...F1
                }}
            }
            fragment F1 on Model1Connection{
                 edges{
                     node{
                         ...F2
                     }
                 }
            }
            fragment F2 on Model1{
                field1
                model2{
                    field2
                }
            }
            `, schema_1.default);
        const fields = parser.getFields();
        expect(fields).toMatchSnapshot();
        expect(parser.getQueryConnectionFields()).toMatchSnapshot();
        const sel = fields[0].fields[0].fields[1].fields[0].node.selectionSet;
        if (sel) {
            sel.selections.push({
                name: { kind: "Name", value: "id" },
                kind: "Field",
            });
        }
        expect(parser.print()).toMatchSnapshot();
    });
    it("mutation payload fields", () => {
        const parser = __1.fromQuery(`mutation M1{
            createModel(input:{ value: "Hello" }){
                clientMutationId
                model {
                    name
                    id
                }
            }
        }`);
        expect(parser.getFields()).toMatchSnapshot();
    });
});
