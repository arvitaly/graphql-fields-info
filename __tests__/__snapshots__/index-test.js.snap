// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info get node fields 1`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [
      Object {
        "args": Array [],
        "fields": Array [],
        "isConnection": false,
        "isNode": false,
        "name": "field1",
        "node": Object {
          "alias": null,
          "arguments": Array [],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 38,
            "start": 32,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 38,
              "start": 32,
            },
            "value": "field1",
          },
          "selectionSet": null,
        },
      },
    ],
    "isConnection": false,
    "isNode": false,
    "name": "model1",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 40,
        "start": 24,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 30,
          "start": 24,
        },
        "value": "model1",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "loc": Object {
          "end": 40,
          "start": 30,
        },
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 38,
              "start": 32,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 38,
                "start": 32,
              },
              "value": "field1",
            },
            "selectionSet": null,
          },
        ],
      },
    },
  },
]
`;

exports[`info get query connection fields simple 1`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [],
    "isConnection": false,
    "isNode": false,
    "name": "name",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 148,
        "start": 144,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 148,
          "start": 144,
        },
        "value": "name",
      },
      "selectionSet": null,
    },
  },
]
`;

exports[`info get query connection fields with fragments, change selection node and print 1`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [
      Object {
        "args": Array [],
        "fields": Array [
          Object {
            "args": Array [],
            "fields": Array [
              Object {
                "args": Array [],
                "fields": Array [],
                "isConnection": false,
                "isNode": false,
                "name": "hasNextPage",
                "node": Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 129,
                    "start": 118,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 129,
                      "start": 118,
                    },
                    "value": "hasNextPage",
                  },
                  "selectionSet": null,
                },
                "type": "Boolean",
              },
            ],
            "isConnection": false,
            "isNode": false,
            "name": "pageInfo",
            "node": Object {
              "alias": null,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 151,
                "start": 84,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 92,
                  "start": 84,
                },
                "value": "pageInfo",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 151,
                  "start": 92,
                },
                "selections": Array [
                  Object {
                    "alias": null,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 129,
                      "start": 118,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 129,
                        "start": 118,
                      },
                      "value": "hasNextPage",
                    },
                    "selectionSet": null,
                  },
                ],
              },
            },
            "type": "Model1ConnectionPageInfo",
          },
          Object {
            "args": Array [],
            "fields": Array [
              Object {
                "args": Array [],
                "fields": Array [
                  Object {
                    "args": Array [],
                    "fields": Array [],
                    "isConnection": false,
                    "isNode": false,
                    "name": "field1",
                    "node": Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 451,
                        "start": 445,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 451,
                          "start": 445,
                        },
                        "value": "field1",
                      },
                      "selectionSet": null,
                    },
                    "type": "String",
                    "typeName": "Model1",
                  },
                  Object {
                    "args": Array [],
                    "fields": Array [
                      Object {
                        "args": Array [],
                        "fields": Array [],
                        "isConnection": false,
                        "isNode": false,
                        "name": "field2",
                        "node": Object {
                          "alias": null,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "loc": Object {
                            "end": 502,
                            "start": 496,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 502,
                              "start": 496,
                            },
                            "value": "field2",
                          },
                          "selectionSet": null,
                        },
                        "type": "Int",
                      },
                    ],
                    "isConnection": false,
                    "isNode": true,
                    "name": "model2",
                    "node": Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 520,
                        "start": 468,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 474,
                          "start": 468,
                        },
                        "value": "model2",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "loc": Object {
                          "end": 520,
                          "start": 474,
                        },
                        "selections": Array [
                          Object {
                            "alias": null,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "loc": Object {
                              "end": 502,
                              "start": 496,
                            },
                            "name": Object {
                              "kind": "Name",
                              "loc": Object {
                                "end": 502,
                                "start": 496,
                              },
                              "value": "field2",
                            },
                            "selectionSet": null,
                          },
                        ],
                      },
                    },
                    "type": "Model2",
                    "typeName": "Model1",
                  },
                ],
                "isConnection": false,
                "isNode": true,
                "name": "node",
                "node": Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 360,
                    "start": 301,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 305,
                      "start": 301,
                    },
                    "value": "node",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "loc": Object {
                      "end": 360,
                      "start": 305,
                    },
                    "selections": Array [
                      Object {
                        "directives": Array [],
                        "kind": "FragmentSpread",
                        "loc": Object {
                          "end": 337,
                          "start": 332,
                        },
                        "name": Object {
                          "kind": "Name",
                          "loc": Object {
                            "end": 337,
                            "start": 335,
                          },
                          "value": "F2",
                        },
                      },
                    ],
                  },
                },
                "type": "Model1",
              },
            ],
            "isConnection": false,
            "isNode": false,
            "name": "edges",
            "node": Object {
              "alias": null,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 379,
                "start": 273,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 278,
                  "start": 273,
                },
                "value": "edges",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 379,
                  "start": 278,
                },
                "selections": Array [
                  Object {
                    "alias": null,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 360,
                      "start": 301,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 305,
                        "start": 301,
                      },
                      "value": "node",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "loc": Object {
                        "end": 360,
                        "start": 305,
                      },
                      "selections": Array [
                        Object {
                          "directives": Array [],
                          "kind": "FragmentSpread",
                          "loc": Object {
                            "end": 337,
                            "start": 332,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 337,
                              "start": 335,
                            },
                            "value": "F2",
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
            "type": "[Model1ConnectionEdge]!",
            "typeName": "Model1Connection",
          },
        ],
        "isConnection": true,
        "isNode": false,
        "name": "model1",
        "node": Object {
          "alias": null,
          "arguments": Array [
            Object {
              "kind": "Argument",
              "loc": Object {
                "end": 61,
                "start": 53,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 58,
                  "start": 53,
                },
                "value": "first",
              },
              "value": Object {
                "kind": "IntValue",
                "loc": Object {
                  "end": 61,
                  "start": 59,
                },
                "value": "10",
              },
            },
          ],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 195,
            "start": 46,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 52,
              "start": 46,
            },
            "value": "model1",
          },
          "selectionSet": Object {
            "kind": "SelectionSet",
            "loc": Object {
              "end": 195,
              "start": 62,
            },
            "selections": Array [
              Object {
                "alias": null,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "loc": Object {
                  "end": 151,
                  "start": 84,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 92,
                    "start": 84,
                  },
                  "value": "pageInfo",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "loc": Object {
                    "end": 151,
                    "start": 92,
                  },
                  "selections": Array [
                    Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 129,
                        "start": 118,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 129,
                          "start": 118,
                        },
                        "value": "hasNextPage",
                      },
                      "selectionSet": null,
                    },
                  ],
                },
              },
              Object {
                "directives": Array [],
                "kind": "FragmentSpread",
                "loc": Object {
                  "end": 177,
                  "start": 172,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 177,
                    "start": 175,
                  },
                  "value": "F1",
                },
              },
            ],
          },
        },
        "type": "Model1Connection",
      },
    ],
    "isConnection": false,
    "isNode": true,
    "name": "viewer",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 196,
        "start": 22,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 28,
          "start": 22,
        },
        "value": "viewer",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "loc": Object {
          "end": 196,
          "start": 28,
        },
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "loc": Object {
                  "end": 61,
                  "start": 53,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 58,
                    "start": 53,
                  },
                  "value": "first",
                },
                "value": Object {
                  "kind": "IntValue",
                  "loc": Object {
                    "end": 61,
                    "start": 59,
                  },
                  "value": "10",
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 195,
              "start": 46,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 52,
                "start": 46,
              },
              "value": "model1",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "loc": Object {
                "end": 195,
                "start": 62,
              },
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 151,
                    "start": 84,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 92,
                      "start": 84,
                    },
                    "value": "pageInfo",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "loc": Object {
                      "end": 151,
                      "start": 92,
                    },
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "loc": Object {
                          "end": 129,
                          "start": 118,
                        },
                        "name": Object {
                          "kind": "Name",
                          "loc": Object {
                            "end": 129,
                            "start": 118,
                          },
                          "value": "hasNextPage",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "directives": Array [],
                  "kind": "FragmentSpread",
                  "loc": Object {
                    "end": 177,
                    "start": 172,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 177,
                      "start": 175,
                    },
                    "value": "F1",
                  },
                },
              ],
            },
          },
        ],
      },
    },
    "type": "Viewer",
  },
]
`;

exports[`info get query connection fields with fragments, change selection node and print 2`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [],
    "isConnection": false,
    "isNode": false,
    "name": "field1",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 451,
        "start": 445,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 451,
          "start": 445,
        },
        "value": "field1",
      },
      "selectionSet": null,
    },
    "type": "String",
    "typeName": "Model1",
  },
  Object {
    "args": Array [],
    "fields": Array [
      Object {
        "args": Array [],
        "fields": Array [],
        "isConnection": false,
        "isNode": false,
        "name": "field2",
        "node": Object {
          "alias": null,
          "arguments": Array [],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 502,
            "start": 496,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 502,
              "start": 496,
            },
            "value": "field2",
          },
          "selectionSet": null,
        },
        "type": "Int",
      },
    ],
    "isConnection": false,
    "isNode": true,
    "name": "model2",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 520,
        "start": 468,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 474,
          "start": 468,
        },
        "value": "model2",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "loc": Object {
          "end": 520,
          "start": 474,
        },
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 502,
              "start": 496,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 502,
                "start": 496,
              },
              "value": "field2",
            },
            "selectionSet": null,
          },
        ],
      },
    },
    "type": "Model2",
    "typeName": "Model1",
  },
]
`;

exports[`info get query connection fields with fragments, change selection node and print 3`] = `
"query Q1 {
  viewer {
    model1(first: 10) {
      pageInfo {
        hasNextPage
      }
      ...F1
    }
  }
}
fragment F1 on Model1Connection {
  edges {
    node {
      ...F2
      id
    }
  }
}
fragment F2 on Model1 {
  field1
  model2 {
    field2
  }
}"
`;

exports[`info get query one fields 1`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [],
    "isConnection": false,
    "isNode": false,
    "name": "field1",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 32,
        "start": 26,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 32,
          "start": 26,
        },
        "value": "field1",
      },
      "selectionSet": null,
    },
  },
  Object {
    "args": Array [],
    "fields": Array [
      Object {
        "args": Array [],
        "fields": Array [],
        "isConnection": false,
        "isNode": false,
        "name": "field2",
        "node": Object {
          "alias": null,
          "arguments": Array [],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 48,
            "start": 42,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 48,
              "start": 42,
            },
            "value": "field2",
          },
          "selectionSet": null,
        },
      },
    ],
    "isConnection": false,
    "isNode": false,
    "name": "model2",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 50,
        "start": 34,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 40,
          "start": 34,
        },
        "value": "model2",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "loc": Object {
          "end": 50,
          "start": 40,
        },
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 48,
              "start": 42,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 48,
                "start": 42,
              },
              "value": "field2",
            },
            "selectionSet": null,
          },
        ],
      },
    },
  },
]
`;

exports[`info parse all fields 1`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [
      Object {
        "args": Array [],
        "fields": Array [
          Object {
            "args": Array [],
            "fields": Array [],
            "isConnection": false,
            "isNode": false,
            "name": "field1",
            "node": Object {
              "alias": null,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 50,
                "start": 44,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 50,
                  "start": 44,
                },
                "value": "field1",
              },
              "selectionSet": null,
            },
          },
        ],
        "isConnection": false,
        "isNode": false,
        "name": "model1",
        "node": Object {
          "alias": null,
          "arguments": Array [],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 52,
            "start": 36,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 42,
              "start": 36,
            },
            "value": "model1",
          },
          "selectionSet": Object {
            "kind": "SelectionSet",
            "loc": Object {
              "end": 52,
              "start": 42,
            },
            "selections": Array [
              Object {
                "alias": null,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "loc": Object {
                  "end": 50,
                  "start": 44,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 50,
                    "start": 44,
                  },
                  "value": "field1",
                },
                "selectionSet": null,
              },
            ],
          },
        },
      },
      Object {
        "args": Array [],
        "fields": Array [
          Object {
            "args": Array [],
            "fields": Array [],
            "isConnection": false,
            "isNode": false,
            "name": "field2",
            "node": Object {
              "alias": null,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 130,
                "start": 124,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 130,
                  "start": 124,
                },
                "value": "field2",
              },
              "selectionSet": null,
            },
          },
        ],
        "isConnection": false,
        "isNode": false,
        "name": "model2",
        "node": Object {
          "alias": null,
          "arguments": Array [],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 162,
            "start": 65,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 71,
              "start": 65,
            },
            "value": "model2",
          },
          "selectionSet": Object {
            "kind": "SelectionSet",
            "loc": Object {
              "end": 162,
              "start": 71,
            },
            "selections": Array [
              Object {
                "directives": Array [],
                "kind": "InlineFragment",
                "loc": Object {
                  "end": 148,
                  "start": 89,
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "loc": Object {
                    "end": 148,
                    "start": 102,
                  },
                  "selections": Array [
                    Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 130,
                        "start": 124,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 130,
                          "start": 124,
                        },
                        "value": "field2",
                      },
                      "selectionSet": null,
                    },
                  ],
                },
                "typeCondition": Object {
                  "kind": "NamedType",
                  "loc": Object {
                    "end": 102,
                    "start": 96,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 102,
                      "start": 96,
                    },
                    "value": "Model2",
                  },
                },
              },
            ],
          },
        },
      },
    ],
    "isConnection": false,
    "isNode": false,
    "name": "node",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 172,
        "start": 18,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 22,
          "start": 18,
        },
        "value": "node",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "loc": Object {
          "end": 172,
          "start": 22,
        },
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 52,
              "start": 36,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 42,
                "start": 36,
              },
              "value": "model1",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "loc": Object {
                "end": 52,
                "start": 42,
              },
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 50,
                    "start": 44,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 50,
                      "start": 44,
                    },
                    "value": "field1",
                  },
                  "selectionSet": null,
                },
              ],
            },
          },
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 162,
              "start": 65,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 71,
                "start": 65,
              },
              "value": "model2",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "loc": Object {
                "end": 162,
                "start": 71,
              },
              "selections": Array [
                Object {
                  "directives": Array [],
                  "kind": "InlineFragment",
                  "loc": Object {
                    "end": 148,
                    "start": 89,
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "loc": Object {
                      "end": 148,
                      "start": 102,
                    },
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "loc": Object {
                          "end": 130,
                          "start": 124,
                        },
                        "name": Object {
                          "kind": "Name",
                          "loc": Object {
                            "end": 130,
                            "start": 124,
                          },
                          "value": "field2",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                  "typeCondition": Object {
                    "kind": "NamedType",
                    "loc": Object {
                      "end": 102,
                      "start": 96,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 102,
                        "start": 96,
                      },
                      "value": "Model2",
                    },
                  },
                },
              ],
            },
          },
        ],
      },
    },
  },
]
`;

exports[`info simple query and print 1`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [],
    "isConnection": false,
    "isNode": false,
    "name": "test",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 26,
        "start": 22,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 26,
          "start": 22,
        },
        "value": "test",
      },
      "selectionSet": null,
    },
    "type": "String",
  },
]
`;

exports[`info simple query and print 2`] = `
"query Q1 {
  test
}
"
`;
